src/dynamics/none/dyn_grid.F90

```diff
@@ -626,4 +634,72 @@ subroutine find_dimension(file, dim_names, found_name, dim_len)
       end if
    end subroutine find_dimension
 
+   !===========================================================================
+
+   subroutine find_energy_formula(file, grid_is_latlon)
+      use pio,                  only: file_desc_t
+      use pio,                  only: pio_inq_att, pio_global, PIO_NOERR
+      use cam_thermo_formula,   only: energy_formula_physics, energy_formula_dycore
+      use cam_thermo_formula,   only: ENERGY_FORMULA_DYCORE_SE, ENERGY_FORMULA_DYCORE_FV, ENERGY_FORMULA_DYCORE_MPAS
+      use physics_types,        only: dycore_energy_consistency_adjust
+      use phys_vars_init_check, only: mark_as_initialized
+
+      ! Find which dynamical core is used in <file> and set the energy formulation
+      ! (also called vc_dycore in CAM)
+
+      type(file_desc_t), intent(inout) :: file
+      logical, intent(in)              :: grid_is_latlon
+
+      ! Local variables
+      integer                          :: ierr, xtype
+      character(len=*), parameter      :: subname = 'find_energy_formula'
+
+      energy_formula_dycore = -1
+
+      ! Is FV dycore? (has lat lon dimension)
+      if(grid_is_latlon) then
```

Comment:
Not sure I understand how this code works but it does not seem general to assume that a lat-lon grid implies a particular energy formula. Is this a replacement for ```is_dycore``` function?

Reply:
Thanks for the question! This is used only for the null dycore to recognize which dycore configuration of CAM created the snapshot being used, so the energy formula used for testing can be determined.

Unfortunately the snapshot metadata does not explicitly specify this, so there is some guesswork here. If the snapshot file has a lat-lon grid `FV` is assumed; if there is a `ne` attribute `SE` is assumed; otherwise `MPAS` is assumed.

I'm happy to revise this however to better recognize this a new attribute probably has to be added to the CAM snapshots, but the 'guesswork' might have to remain for the existing/old snapshots.